src/MICore/TerminalLauncher.cs

```diff
@@ -106,7 +95,54 @@ protected override string GetProcessArgs()
                 throw new FileNotFoundException(message);
             }
 
-            return string.Format(CultureInfo.InvariantCulture, "{0} \"{1}\" \"{2}\"", launchScript, _title, _initScript);
+            string setEnvironmentScript = GetSetEnvironmentScript();
+            return string.Format(CultureInfo.InvariantCulture, "{0} \"{1}\" \"{2}{3}\"", launchScript, _title, setEnvironmentScript, _initScript);
+        }
+
+        private string GetSetEnvironmentScript()
+        {
+            if (_environment == null)
+            {
+                return string.Empty;
+            }
+
+            StringBuilder stringBuilder = new StringBuilder();
+            foreach (EnvironmentEntry entry in _environment)
+            {
+                stringBuilder.Append("export ");
+                stringBuilder.Append(entry.Name);
+                stringBuilder.Append("='");
+
+                // If the value doesn't contain single quote characters ('), we can just drop it in
+                if (entry.Value.IndexOf('\'') < 0)
+                {
+                    stringBuilder.Append(entry.Value);
+                }
+                else
+                {
+                    // otherwise we need to escape the single quotes
+                    foreach (char c in entry.Value)
+                    {
+                        if (c != '\'')
+                        {
+                            stringBuilder.Append(c);
+                        }
+                        else
+                        {
+                            stringBuilder.Append("'\\\''"); // end the single quoted string we are in, then add a single quote, and restart the single quotes
```

Comment:
I updated the comment so hopefully it is more clear. I also used a '@' string which hopefully also helps.